libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp

```diff
@@ -240,6 +240,9 @@ AP_AdvancedFailsafe::check(uint32_t last_valid_rc_ms)
             if (_wp_comms_hold) {
                 _saved_wp = mission.get_current_nav_cmd().index;
                 mission.set_current_cmd(_wp_comms_hold);
+                if (mode == AFS_AUTO) {
```

Comment:
I think we need an AFS_OPTIONS bitmask (an AP_Int16) which has an option to enable this new behaviour. Some users may be relying on the current behaviour, eg. if they takeover with an RC switch to force QLAND mode, then find they keep going back to AUTO

Reply:
Thanks @tridge 
I'll get back to this PR once #23334 is sorted out. I just added AFS_OPTIONS in that PR. That PR is a bit simpler than this one.

Reply:
AFS_OPTIONS for force entering AUTO is now added.